x/bank/src/keeper.rs

```diff
@@ -80,40 +93,122 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, AK: AuthKeeper<SK, M>, M: Module> Ban
             }))
     }
 
-    fn balance_all<DB: Database, CTX: QueryableContext<DB, SK>>(
+    fn coins_burn<DB: Database, CTX: TransactionalContext<DB, SK>>(
         &self,
-        _ctx: &CTX,
-        _address: &AccAddress,
-    ) -> Result<Vec<Coin>, GasStoreErrors> {
-        unimplemented!() // TODO:NOW IMPLEMENT THIS ONE
+        ctx: &mut CTX,
+        module: &M,
+        deposit: &SendCoins,
+    ) -> Result<(), AppError> {
+        let module_acc_addr = module.get_address();
+
+        let account = self
+            .auth_keeper
+            .get_account(ctx, &module_acc_addr)?
+            .unwrap(); // TODO:
+
+        match account.has_permissions("burner") {
```

Comment:
We should try to move away from using Strings for permissions.

Reply:
True, but I don't want to resolve this it in this PR